### PR TITLE
Find the number of all possible binary search trees

### PR DESCRIPTION
**Pull request by alpeshyadav**
### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: data_structures/binary_tree/binary_search_tree.py [hunk 1]
```diff
@@ -155,6 +155,15 @@ def find_kth_smallest(self, k: int, node: Node) -> int:
...
         return arr[k - 1]
 
+    def total_possible_BST(self, node:Node) -> int:
+        from math import factorial
+        arr = []
+        self.inorder(arr, node)
+        n = len(arr)
+        # Catalan number Cn = (2n)! / ((n + 1)! * n!)
+        return factorial(2 * n) // factorial(n + 1) // factorial(n)
+
```
Comment (cclauss) [inline data_structures/binary_tree/binary_search_tree.py:165]:
```suggestion
    def total_possible_binary_search_trees(self, node:Node) -> int:
        arr = []
        self.inorder(arr, node)
        n = len(arr)
        # Catalan number Cn = (2n)! / ((n + 1)! * n!)
        return factorial(2 * n) // factorial(n + 1) // factorial(n)
```

Comment (cclauss) [reply data_structures/binary_tree/binary_search_tree.py:165]: Please move the import to the top of the file.

Comment (alpeshyadav) [reply data_structures/binary_tree/binary_search_tree.py:165]: okay

File: data_structures/binary_tree/binary_search_tree.py [hunk 2]
```diff
@@ -152,26 +154,26 @@ def inorder(self, arr: list, node: Node):
     def find_kth_smallest(self, k: int, node: Node) -> int:
         """Return the kth smallest element in a binary search tree """
         arr = []
-        self.inorder(arr, node)  # append all values to list using inorder traversal
+        # append all values to list using inorder traversal
```
Comment (cclauss) [inline data_structures/binary_tree/binary_search_tree.py:156]: Why??

File: data_structures/binary_tree/binary_search_tree.py [hunk 3]
```diff
@@ -152,26 +154,26 @@ def inorder(self, arr: list, node: Node):
...
 
-
 def postorder(curr_node):
     """
     postOrder (left, right, self)
     """
     node_list = list()
     if curr_node is not None:
-        node_list = postorder(curr_node.left) + postorder(curr_node.right) + [curr_node]
+        node_list = postorder(curr_node.left) + \
```
Comment (cclauss) [inline data_structures/binary_tree/binary_search_tree.py:174]: Please undo this useless change.  PEP8 cautions us not to use backslashes in Python code.

File: data_structures/binary_tree/binary_search_tree.py (hunk 2 above)
Comment (alpeshyadav) [reply data_structures/binary_tree/binary_search_tree.py:156]: Oh, it happened by mistake.

File: data_structures/binary_tree/binary_search_tree.py (hunk 3 above)
Comment (alpeshyadav) [reply data_structures/binary_tree/binary_search_tree.py:174]: Sure, I'm using black formatter, it did all these changes

**Comment by alpeshyadav** [comment]
should I need to change anything for merging?

**Comment by alpeshyadav** [comment]
How to trace it? I tried and the error is showing in the postorder function which I didn't altered.

**Pull request closed without merging**